src/relayclient/Enveloping.ts

```diff
@@ -0,0 +1,236 @@
+    * @param  recoverer optional: This SmartWallet instance won't have recovery support
+    * @return a deploy request structure.
+    */
+  async createDeployRequest (from: Address, gasLimit: IntString, tokenContract: Address, tokenAmount: IntString, tokenGas: IntString, gasPrice?: IntString, index? : IntString, recoverer? : IntString): Promise<DeployRequest> {
+    const deployRequest: DeployRequest = {
+      request: {
+        relayHub: this.config.relayHubAddress,
+        from: from,
+        to: zeroAddr,
+        value: '0',
+        gas: gasLimit, // overhead (cte) + fee + (estimateDeploy * 1.1)
+        nonce: (await this.getFactoryNonce(this.config.smartWalletFactoryAddress, from)).toString(),
+        data: '0x',
+        tokenContract: tokenContract,
+        tokenAmount: tokenAmount,
+        tokenGas: tokenGas,
+        recoverer: recoverer ?? constants.ZERO_ADDRESS,
+        index: index ?? '0'
+      },
+      relayData: {
+        gasPrice: gasPrice ?? '0',
+        relayWorker: this.relayWorkerAddress,
+        callForwarder: this.config.smartWalletFactoryAddress,
+        callVerifier: this.config.deployVerifierAddress,
+        domainSeparator: getDomainSeparatorHash(this.config.smartWalletFactoryAddress, this.config.chainId)
+      }
+    }
+
+    return deployRequest
+  }
+
+  /**
+    * creates a relay request
+    * @param from - sender's wallet address (EOA)
+    * @param to - recipient contract
+    * @param forwarder - smart wallet address forwarding the relay request
+    * @param data - payload for the execution (e.g. encoded call when invoking an smart contract)
+    * @param gasLimit - gas limit of the relayed transaction
+    * @param tokenContract - the token the user will use to pay to the Worker for relaying
+    * @param tokenAmount - the token amount the user will pay for relaying, zero if the call is subsidized
+    * @param tokenGas - gas limit of the token payment
+    * @param  gasPrice - optional: if not set the gasPrice is calculated internally through a call to estimateGas
+    * @return a relay request structure.
+    */
+  async createRelayRequest (from: Address, to: Address, forwarder: Address, data: PrefixedHexString, gasLimit: IntString, tokenContract: Address, tokenAmount: IntString, tokenGas: IntString, gasPrice?: IntString): Promise<RelayRequest> {
+    const relayRequest: RelayRequest = {
+      request: {
+        relayHub: this.config.relayHubAddress,
+        from: from,
+        to: to,
+        data: data,
+        value: '0',
+        gas: gasLimit,
+        nonce: (await this.getSenderNonce(forwarder)).toString(),
+        tokenContract: tokenContract,
+        tokenAmount: tokenAmount,
+        tokenGas: tokenGas
+      },
+      relayData: {
+        gasPrice: gasPrice ?? '0',
+        relayWorker: this.relayWorkerAddress,
+        callForwarder: forwarder,
+        callVerifier: this.config.relayVerifierAddress,
```

Comment:
relayVerifierAddress is for the "simple version".  This cannot be used then for custom smart wallets